packages/create-tldraw/src/main.ts

```diff
@@ -0,0 +1,232 @@
+	await renameTemplate(name, targetDir)
+
+	const manager = getPackageManager()
+	const doneMessage = ['Done! Now run:', '']
+	if (targetDir !== process.cwd()) {
+		doneMessage.push(`   cd ${relative(process.cwd(), targetDir)}`)
+	}
+	doneMessage.push(`   ${getInstallCommand(manager)}`)
+	doneMessage.push(`   ${getRunCommand(manager, 'dev')}`)
+
+	outro(doneMessage.join('\n'))
+}
+
+main().catch((err) => {
+	if (DEBUG) console.error(err)
+	outro(`it's bad`)
+	process.exit(1)
+})
+
+async function templatePicker(argOption?: string) {
+	if (argOption) {
+		const template = allTemplates.find((t) => {
+			return (
+				t.repo.replace(/^tldraw\//, '').toLowerCase() === argOption.toLowerCase() ||
+				t.name.toLowerCase() === argOption.toLowerCase()
+			)
+		})
+
+		if (!template) {
+			outro(`Template ${argOption} not found`)
+			process.exit(1)
+		}
+
+		return template
+	}
+
+	function formatTemplates(templates: Template[], groupLabel: string) {
+		return templates.map((template) => ({
+			label: template.name,
+			hint: template.description,
+			value: template,
+			group: groupLabel,
+		}))
+	}
+
+	return await handleCancel(
+		groupSelect({
+			message: 'Select a template',
+			options: [
+				...formatTemplates(TEMPLATES.framework, 'Frameworks'),
+				...formatTemplates(TEMPLATES.app, 'Apps'),
+			],
+		})
+	)
+}
+
+async function namePicker(argOption?: string) {
+	if (argOption) {
+		return pathToName(argOption)
+	}
+
+	const defaultName = pathToName(process.cwd())
+
+	const name = await handleCancel(
+		text({
+			message: picocolors.bold('Name your package'),
+			placeholder: defaultName,
+			validate: (value) => {
+				if (value && !isValidPackageName(value)) {
+					return `Invalid package name: ${value}`
+				}
+
+				return undefined
+			},
+		})
+	)
+
+	if (!name.trim()) return defaultName
+	return pathToName(name)
+}
+
+async function ensureDirectoryEmpty(targetDir: string, overwriteArg: boolean) {
+	if (isDirEmpty(targetDir)) {
+		mkdirSync(targetDir, { recursive: true })
+		return
+	}
+
+	const overwrite = overwriteArg
+		? 'yes'
+		: await select({
+				message: picocolors.bold(
+					(targetDir === process.cwd()
+						? 'Current directory'
+						: `Target directory "${path.relative(process.cwd(), targetDir)}"`) + ` is not empty.`
+				),
+				options: [
+					{
+						label: 'Cancel',
+						value: 'no',
+					},
+					{
+						label: 'Remove existing files and continue',
+						value: 'yes',
+					},
+					{
+						label: 'Ignore existing files and continue',
+						value: 'ignore',
+					},
+				],
+			})
+
+	if (overwrite === 'no') {
+		outro(`it's cancelled`)
+		process.exit(1)
+	}
+
+	if (overwrite === 'yes') {
+		emptyDir(targetDir)
+	}
+}
+
+async function downloadTemplate(template: Template, targetDir: string) {
+	const s = spinner()
+	s.start(`Downloading github.com/${template.repo}...`)
+	try {
+		const url = `https://github.com/${template.repo}/archive/refs/heads/main.tar.gz`
```

Comment:
oo neat

Reply:
actually, checking here, if this is pulling from main, possible that there's a version skew / types might be off until the tldraw version is matched with the template?

actually, for my understanding, how does this handle the `tldraw` version as they're currently set to `workspace:*`?

Reply:
nope - this pulls from the published template repositories, not our monorepo

Reply:
ah right right, i see it now